zebra-state/src/service/check/tests/anchors.rs

```diff
@@ -0,0 +1,116 @@
+use std::{convert::TryInto, ops::Deref, sync::Arc};
+
+use zebra_chain::{
+    block::{Block, Height},
+    serialization::ZcashDeserializeInto,
+    transaction::{LockTime, Transaction},
+};
+
+use crate::{
+    arbitrary::Prepare,
+    tests::setup::{new_state_with_mainnet_genesis, transaction_v4_from_coinbase},
+};
+
+// sapling
+
+/// Check that, when primed with the first Sapling blocks, a Sapling Spend's referenced anchor is
+/// validated.
+#[test]
+fn check_sapling_anchors() {
```

Comment:
As I understand this tests only the non-finalized state since it only works with two blocks.
It would be good to check if it works if the anchor is in the finalized state. Not sure what's the easiest way - maybe doing a similar test with ~100 (I forget the exact number) blocks in between those two blocks, or using a `PreparedChain` proptest.

Reply:
Thinking about it, since you've managed to run this to the tip, this was already indirectly tested.